### PR TITLE
loader: flag that does not put nil values in the cache

### PR DESCRIPTION
**Pull request by gazz**
@skazhy @axelarge what do you think?

### CODE REVIEW

File: src/full/cache.clj [hunk 1]
```diff
@@ -60,7 +60,9 @@
                     v (if (instance? ReadPort r)
                         (<! r)
                         r)]
-                (when (not (instance? Throwable v))
+                (when (and (not (instance? Throwable v))
+                           (or v
```
Comment (vindavs) [inline src/full/cache.clj]: This should check `(some? v)`, otherwise `false` also won't get stored

File: src/full/cache.clj [hunk 2]
```diff
@@ -60,7 +60,9 @@
...
+                               (and (not v) allow-nil?)))
```
Comment (vindavs) [inline src/full/cache.clj]: `(not v)` is superfluous here

**Review by vindavs — CHANGES_REQUESTED** [review]

File: src/full/cache.clj [hunk 3]
```diff
@@ -46,7 +46,7 @@
     ::none))
 
 (defn- do-get-or-load>
-  [k loader> timeout {:keys [setf getf states]}]
+  [k loader> timeout {:keys [setf getf states allow-nil?]}]
```
Comment (vindavs) [inline src/full/cache.clj:49]: A default of `true` needs to be added here or in `lget-or-load>`, otherwise `lget-or-load>` changes functionality.

**Comment by gazz** [comment]
another pass lielo cirvīt?

**Review by vindavs — APPROVED** [review]

**Pull request merged**